src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3509,7 +3514,6 @@ private BoundExpression BindClassCreationExpression(ObjectCreationExpressionSynt
                 }
                 else if (node.Type.Kind() == SyntaxKind.TupleType)
                 {
-                    Debug.Assert(node.HasErrors, "new <tuple type> should be a syntax error");
                     return MakeBadExpressionForObjectCreation(node, type, boundInitializerOpt, analyzedArguments);
```

Comment:
Consider adding the diagnostic here rather than in `BindObjectCreationExpression`, so it's clear the bad expression node has a corresponding diagnostic.